src/main/java/pro/belbix/ethparser/web3/Web3Service.java

```diff
@@ -408,6 +408,23 @@ public void subscribeTransactionFlowable() {
         log.info("Subscribe to Transaction Flowable");
     }
 
+    public void subscribeTransactionFlowableRange(Integer start, Integer end)
+    {
+        checkInit();
+        Flowable<Transaction> flowable;
+        log.info("Start flow for block range " + start + " - " + end);
+        flowable = callWithRetry(() -> web3.replayPastTransactionsFlowable(
+                DefaultBlockParameter.valueOf(new BigInteger(start.toString())),
+                DefaultBlockParameter.valueOf(new BigInteger(end.toString()))));
+        Disposable subscription = flowable
+                .subscribe(tx -> transactionConsumers.forEach(queue ->
+                                writeInQueue(queue, tx)),
+                        e -> log.error("Transaction flowable error", e));
+        subscriptions.add(subscription);
```

Comment:
this solution will be unclear and hard to use
add queue as an argument
return Disposable

Reply:
Ok, working on some refactoring for this